Feature/update main

Cyclic opened the pull request:
- Added better orientation change animation and button layout for orientation changes, added exif and error output, added error handling for cases where exif data is not present after capture, add check that existing capture is not taking place, updated constraints to handle more scenarios when the layout changes, added a scale option to use scale by a factor to output of new photos. Added a few more options for output.

Some known layout issues with the main viewController that should be easy to resolve.

AlexLittlejohn commented:
I'll take a look at this shortly.

AlexLittlejohn commented:
This is a huge change that introduces a great many dependancies along with other changes. This renders it unusable for most people as the cost of integrating it would then be too high. If you want you can continue to use your fork and possibly submit more isolated changes?

Cyclic commented:
Thank you Alex. I will remove the cocoa pods from the project, they’re unused. It’s a swift 4 build and there shouldn’t be any issues for existing users if they use the existing plugin. The main changes are cosmetic with a few new return types. I also did some linting that was needed. Would you mind giving it a run in Xcode 9 and commenting on the major breaking changes besides swift? I don’t think the fork has as many differences as you might think. 

Sent from my iPhone

> On Nov 8, 2017, at 5:53 AM, Alex Littlejohn <notifications@github.com> wrote:
> 
> This is a huge change that introduces a great many dependancies along with other changes. This renders it unusable for most people as the cost of integrating it would then be too high. If you want you can continue to use your fork and possibly submit more isolated changes?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

Cyclic commented:
@AlexLittlejohn I reverted back to swift 4 for now, there were only a few lines to change. I've also removed all of the cocoapods since they were unnecessary. There is a minor issue on the front screen after the photo is accepted, but it should be pretty straight forward to resolve. The main thing is the linting, the animation on the camera buttons as you rotate the screen (we can make the animation style configurable or you can change it inline), center the close button (lots of feedback about this from users), the additional scale parameter for photo output, and the exif and error data, which help a lot for debugging or logging. The fact that the buttons are pinned on the camera vc, IMO, feels like a more native experience and removes some of the complexity around the constraints.

Cyclic commented:
@AlexLittlejohn I should also mention that I am creating a Cordova iOS plugin to wrap the camera, so I was hoping to be able to pull from your repo.
